### DISCUSSION TITLE
How does {terra} use GDAL?

### DISCUSSION THREAD

**Discussion by joshualerickson** [gdal4al/gdal-examples#4] [Q&A] [answered]
Very noob question but I was curious how GDAL is used in the {terra} package in R?

For example,

```
library(rast)
elev <- terra::rast('/vsicurl/https://prd-tnm.s3.amazonaws.com/StagedProducts/Elecation/13/TIFF/USGS_Seamless_DEM_13.vrt')

```

Pretty cool and crazy that this works somehow even if I don't have GDAL installed on my machine! How the heck does this happen? 

@mdsumner

**Comment by mdsumner** [comment]
on windows and macos cran bundles gdal completely into the package. it's changed over time, but now rtools43 has these libs all collected together in a huge install, each package links to and copies in the DLLs it needs from rtools, so as a .zip it becomes standalone. 

terra uses GDAL very deeply, it's modelled around it and mostly terra objects are opaque pointers to structures in GDAL, GEOS, or PROJ. That's part of the reason why it's so fast

**Reply by joshualerickson** [reply]
Thanks! Ok this is making more sense. Is this the same for plugins? Like the new ESRI gdb raster or MrSID?

**Reply by mdsumner** [reply, marked as answer]
yes I expect mrsid isn't built in to rtools, they're all compiled from scratch by cran 

but, mrsid is a GDAL driver requiring extra software downloaded,  gdb raster is a built-in GDAL driver - plugin means something specific for GDAL itself
